webapp/channels/src/actions/views/create_comment.tsx

```diff
@@ -63,10 +72,27 @@ export function submitPost(channelId: string, rootId: string, draft: PostDraft):
             pending_post_id: `${userId}:${time}`,
             user_id: userId,
             create_at: time,
-            metadata: {},
+            metadata: {...draft.metadata},
             props: {...draft.props},
         } as unknown as Post;
 
+        const channel = getChannel(state, channelId);
```

Comment:
A lot of this was part of the component code. Moving it to the action serves the following purpose:
- We access the store only when we need the information
- Since we are not listening to so many things, we are calculating way less selectors on every state change, and potentially reducing the re-renders.

I feel we should try to promote this approach more often, since I feel we listen to many things from the store to pass them as action arguments, and we may be able to avoid some of those.

Reply:
Oh yeah, I very much agree with that. I think I've had similar comments to that elsewhere here.

I was asking to see if it was broken previously since I wouldn't have been surprised if the code for disabling mention highlights or group mentions just didn't work in the RHS before 😛

Reply:
Oh, the part about disabling mention highlights was a mess. For posts was done in one place, for comments in a completely different place. Fun for all the family XD